### DISCUSSION TITLE
Also having problems with connecting to internet

### DISCUSSION THREAD

**Discussion by SugarRayLua** [utmapp/UTM#2687] [UTM Help (iOS)]
Hi. I'm a newbie to UTM having previously used iSH on my new iPad Pro (currently running iOS 14.7.1, 256GB storage). I am not jailbroken so downloaded UTM.SE.ipa and was able to install it fine using Singulous. The app opens and runs well, and I'm able to download Debian VM from the galley which function fine in terms of using the packages and features that are installed on them. However, I cannot seem to get the internet working on either Debian VM (e.g. LXDE version or XFCE version). No matter what I try on the LXDE VM, the app indicates I have no connection. When I use the XFCE VM, it says I have an Ethernet connection (and can confirm with ip a). However, when I use XFCE VM to try and pull up google or apt install additional Debian packages, the VM says it can't connect to the server. I've tried what others have suggested on GitHub and web to include manually adding both in UTM setting and/or in Linux itself the DNS servers 1.1.1.1, 8.8.4.4 and 8.8.8.8. I also tried reconfiguring DHCP settings in the "advanced options" part of the UTM settings that people have suggested but that did not help either. My iOS settings app indicates that UTM is setup to share internet with the iPad Pro. 

Any suggestions? UTM looks like a great app but limited if unable to access the internet and install additional packages.

Thanks!

Sincerely,

Mike

**Comment by SugarRayLua** [comment]
Dear UTM Community,

I did manage to figure out how to get the internet working (at least for the prebuilt Debian XFCE and Windows 7 VM). On the UTM main screen, under the "network" options, I went clicked on the "show advanced options" and filled in the following:

Guest Network:  10.0.2.2/24

Host Address:  10.0.2.2

DHCP Start:  10.0.2.15

DNS Sever: 1.1.1.1

I left the rest blank, and it worked. 

I hope this might help others trying to install the prebuilt VMs.

Fyi and have a good night,

Mike

**Reply by kmeisthax** [reply]
This half-worked; my Debian VM can now ping the host but it can't do much else.

**Reply by kmeisthax** [reply]
Ok, so it turns out both ping AND DNS is just broken on iOS, at least using the latest UTM on the jailbreak repo on top of iOS 14.5. TCP and UDP passes through just fine.

(I should also point out that I need to `sudo dhclient` every time I start the Debian VM up...)

I worked around the lack of DNS by installing `dnscrypt-proxy` and changing my `/etc/resolv.conf` to hit it. This uses DNS-over-HTTPS which does work even with whatever the heck iOS is doing to UTM's network access. Note that the default behavior of `dnscrypt-proxy` is to listen on `127.0.2.1` so you'll need to make sure that *specific* address is in your `/etc/resolv.conf`.

(For the record, I got that address by following step 6 of https://www.linuxbabe.com/debian/set-up-dns-over-https-doh-resolver-debian-dnsdist )

Anyone following this from a Google search is probably wondering, "wait how do I install `dnscrypt-proxy` if I don't have DNS"? I bootstrapped around this by manually adding `/etc/hosts` entries for every DNS lookup that failed when trying to `apt-get`. AFAIK this was just `deb.debian.org`.
